Sources/CryptoSwift/BlockMode/CFB.swift

```diff
@@ -21,20 +21,29 @@ public struct CFB: BlockMode {
     /// Invalid IV
     case invalidInitializationVector
   }
+    
+  public enum SegmentSize: Int {
+    case cfb8 = 1 // Encrypt byte per byte
+    case cfb128 = 16 // Encrypt 16 bytes per 16 bytes (default)
+  }
 
   public let options: BlockModeOption = [.initializationVectorRequired, .useEncryptToDecrypt]
   private let iv: Array<UInt8>
+  private let segmentSize: SegmentSize
+  public let customBlockSize: Int?
 
-  public init(iv: Array<UInt8>) {
+  public init(iv: Array<UInt8>, segmentSize: SegmentSize = .cfb128) {
     self.iv = iv
+    self.segmentSize = segmentSize
+    self.customBlockSize = segmentSize.rawValue
   }
 
   public func worker(blockSize: Int, cipherOperation: @escaping CipherOperationOnBlock, encryptionOperation: @escaping CipherOperationOnBlock) throws -> CipherModeWorker {
-    if self.iv.count != blockSize {
+    if !(self.iv.count == blockSize || (segmentSize == .cfb8 && self.iv.count == AES.blockSize)) {
```

Comment:
`BlockMode` should stay cipher-independent whenever possible.  Can we get rid of `AES.blockSize` here?

Reply:
The problem here is that `blockSize` is 1 when `segmentSize` is `.cfb8`, and the IV should stay 16 bytes long (for AES), so I can't compare `iv.count` to `blockSize`. Maybe we could add a blockCipherSize arg to know if it's AES or something else? (or keep it like that)

Reply:
hm... why do we need `BlockMode.customBlockSize`? the block size is a parameter already. I don't see much value in this property - I don't see a relation to segment size hm...

how about remove `customBlockMode`, instead store block mode in `CFB` and use the stored value here?

Reply:
I need this constant in the `AES.swift` file for example, because CFB8 is not working on 16 bytes like default AES but 1 byte instead (else there are errors when the entry size is not a multiple of 16 bytes)
https://github.com/krzyzanowskim/CryptoSwift/blob/085709b6a4e33b60fa0b4a4629655fdd1ba5644e/Sources/CryptoSwift/AES.swift#L499

Reply:
this value goes down to block mode anyway, so it can be checked here and leave the block mode itself unchanged. Another thing, in CFB, let's check if passed `blockSize` is 1 byte instead as it's the only compatible.

Reply:
If I remove the `customBlockSize` constant, test fails:
```
Test Case '-[CryptoSwiftTests.AESTests testAESEncryptCFB8]' started.
/Users/nathanfallet/git/CryptoSwift/Tests/CryptoSwiftTests/AESTests.swift:174: error: -[CryptoSwiftTests.AESTests testAESEncryptCFB8] : XCTAssertEqual failed: ("[59, 121, 66]") is not equal to ("[59, 121, 66, 76, 156, 13, 212, 54, 186, 206, 158, 14, 212, 88, 106, 79, 50, 185]") - encryption failed
/Users/nathanfallet/git/CryptoSwift/Tests/CryptoSwiftTests/AESTests.swift:176: error: -[CryptoSwiftTests.AESTests testAESEncryptCFB8] : XCTAssertEqual failed: ("[107]") is not equal to ("[107, 193, 190, 226, 46, 64, 159, 150, 233, 61, 126, 17, 115, 147, 23, 42, 174, 45]") - decryption failed
```
So this constant is necessary for CFB8 to work, here in `AES`'s `Cryptors` extension:

https://github.com/krzyzanowskim/CryptoSwift/pull/855/files#diff-68bfbf24a9e4229282e0d21efc2b8afb5d4d3db5ad131cc6a185d613f0da78e8R20
https://github.com/krzyzanowskim/CryptoSwift/pull/855/files#diff-68bfbf24a9e4229282e0d21efc2b8afb5d4d3db5ad131cc6a185d613f0da78e8R29

Reply:
ok, let's get it in, and I want to play with it a bit, maybe I'll figure how to make it out of it. Great work btw.